autometa/taxonomy/ncbi.py

```diff
@@ -364,60 +347,68 @@ def parse_merged(self):
             logger.debug('merged loaded')
         return merged
 
-    def is_common_ancestor(self, parent_taxid, child_taxid):
-        """Determines whether the provided taxids have a non-root common ancestor
+    def is_common_ancestor(self, taxid_A, taxid_B):
+        """
+        Determines whether the provided taxids have a non-root common ancestor
 
         Parameters
         ----------
-        parent_taxid : int
-            Description of parameter `parent_taxid`.
-        child_taxid : int
-            Description of parameter `child_taxid`.
+        taxid_A : int
+            identifer for a taxon in the Taxonomy Database by NCBI
+        taxid_B : int
+            identifer for a taxon in the Taxonomy Database by NCBI
 
         Returns
         -------
         boolean
-            True if taxids share a common ancestor and False otherwise
+            True if taxids share a common ancestor else False
         """
-        ancestor_taxid = child_taxid
-        while ancestor_taxid != 1:
-            if parent_taxid == ancestor_taxid:
-                return True
-            ancestor_taxid = self.parent(ancestor_taxid)
-        return False
+        taxid_A_lineage_taxids = {taxids.get('taxid')
+                                  for taxids in self.lineage(taxid_A)}
+        taxid_B_lineage_taxids = {taxids.get('taxid')
+                                  for taxids in self.lineage(taxid_B)}
+        common_ancestor = (taxid_B_lineage_taxids.intersection(
+            taxid_A_lineage_taxids))
```

Comment:
Is there a reason for the extra parentheses?

```suggestion
        common_ancestor = taxid_B_lineage_taxids.intersection(taxid_A_lineage_taxids)
```

You could also use the intersection operator... Although I think the `intersection` method is probably better because it is more clear.

```suggestion
        common_ancestor = taxid_B_lineage_taxids & taxid_A_lineage_taxids
```